app/components/Views/AccountConnect/AccountConnectMultiSelector/AccountConnectMultiSelector.tsx

```diff
@@ -31,79 +28,65 @@ import {
-  onPrimaryActionButtonPress,
 }: AccountConnectMultiSelectorProps) => {
   const { styles } = useStyles(styleSheet, { isRenderedAsBottomSheet });
-  const { navigate } = useNavigation();
   const [screen, setScreen] = useState<AccountConnectMultiSelectorScreens>(
     AccountConnectMultiSelectorScreens.AccountMultiSelector,
   );
-  const sortedSelectedAddresses = [...selectedAddresses].sort((a, b) =>
-    a.localeCompare(b),
-  );
-  const [originalSelectedAddresses] = useState<string[]>(
-    sortedSelectedAddresses,
-  );
+
+  const [selectedAddresses, setSelectedAddresses] = useState<string[]>([]);
+
+  useEffect(() => {
+    setSelectedAddresses(defaultSelectedAddresses);
+  }, [
+    setSelectedAddresses,
+    defaultSelectedAddresses,
+  ]);
 
   const onSelectAccount = useCallback(
     (accAddress: string) => {
-      const selectedAddressIndex = selectedAddresses.indexOf(accAddress);
-      // Reconstruct selected addresses.
-      const newAccountAddresses = accounts.reduce((acc, { address }) => {
-        if (accAddress === address) {
-          selectedAddressIndex === -1 && acc.push(address);
-        } else if (selectedAddresses.includes(address)) {
-          acc.push(address);
-        }
-        return acc;
-      }, [] as string[]);
-      onSelectAddress(newAccountAddresses);
+      const updatedSelectedAccountAddresses = selectedAddresses.filter(
+        (selectedAccountId) => !isEqualCaseInsensitive(selectedAccountId, accAddress),
+      );
+
+      if (
+        updatedSelectedAccountAddresses.length === selectedAddresses.length
+      ) {
+        setSelectedAddresses([...selectedAddresses, accAddress]);
+      } else {
+        setSelectedAddresses(updatedSelectedAccountAddresses);
+      }
     },
-    [accounts, selectedAddresses, onSelectAddress],
+    [selectedAddresses, setSelectedAddresses],
   );
 
-  const onRevokeAllHandler = useCallback(async () => {
-    await Engine.context.PermissionController.revokeAllPermissions(hostname);
-    navigate('PermissionsManager');
-  }, [hostname, navigate]);
+  const handleSubmit = useCallback(() => {
+    onSubmit(selectedAddresses);
+  }, [onSubmit, selectedAddresses]);
 
-  const toggleRevokeAllAccountPermissionsModal = useCallback(() => {
-    navigate(Routes.MODAL.ROOT_MODAL_FLOW, {
-      screen: Routes.SHEET.REVOKE_ALL_ACCOUNT_PERMISSIONS,
-      params: {
-        hostInfo: {
-          metadata: {
-            origin: urlWithProtocol && new URL(urlWithProtocol).hostname,
-          },
-        },
-        onRevokeAll: !isRenderedAsBottomSheet && onRevokeAllHandler,
-      },
-    });
-  }, [navigate, urlWithProtocol, isRenderedAsBottomSheet, onRevokeAllHandler]);
+  const handleDisconnect = useCallback(() => {
+    onSubmit([]);
+  }, [onSubmit]);
 
   const areAllAccountsSelected = accounts
-    .map(({ address }) => address)
-    .every((address) => selectedAddresses.includes(address));
+    .every(({address}) => selectedAddresses.includes(address));
```

Comment:
```suggestion
    .every(({ address }) => selectedAddresses.includes(address));
```

so someone is not hit with the "formatting" hammer in the future 🔨